### PR TITLE
Allow popups to be opened for editable notation elements

### PR DESCRIPTION
**Pull request by sidharth-anand**
image

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [ ] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [ ] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

File: src/engraving/libmscore/tempotext.h [hunk 1]
```diff
@@ -83,6 +83,8 @@ class TempoText final : public TextBase
     bool setProperty(Pid propertyId, const QVariant&) override;
     QVariant propertyDefault(Pid id) const override;
     QString accessibleInfo() const override;
+
+    virtual bool needsElementPopup() const override {return true;}
```
Comment (njvdberg) [inline src/engraving/libmscore/tempotext.h:87]: Is the `virtual` keyword required since the class is `final`?

File: src/main/CMakeLists.txt [hunk 2]
```diff
@@ -515,8 +515,5 @@ if (MSVC)
    if (NOT ${CMAKE_VERSION} VERSION_LESS "3.8.0")
       set_target_properties(${EXECUTABLE_NAME} PROPERTIES VS_DEBUGGER_WORKING_DIRECTORY "${VS_DEBUGGER_WORKING_DIRECTORY}")
    endif ()
-   if (NOT ${CMAKE_VERSION} VERSION_LESS "3.13.0")
-      set_target_properties(${EXECUTABLE_NAME} PROPERTIES VS_DEBUGGER_COMMAND_ARGUMENTS "--debug")
-   endif ()
```
Comment (njvdberg) [inline src/main/CMakeLists.txt:520]: Why this change?

File: src/notation/qml/MuseScore/NotationScene/NotationView.qml [hunk 3]
```diff
@@ -23,9 +23,10 @@ import QtQuick 2.15
 import QtQuick.Layouts 1.15
 import QtQuick.Controls 2.15
 
-import MuseScore.Ui 1.0
-import MuseScore.UiComponents 1.0
 import MuseScore.NotationScene 1.0
+import MuseScore.UiComponents 1.0
+import MuseScore.Ui 1.0
```
Comment (njvdberg) [inline src/notation/qml/MuseScore/NotationScene/NotationView.qml:28]: Why are lines 26 and 27 exchanged. Try not to make unnecessary changes.

File: src/notation/view/notationpaintview.cpp [hunk 4]
```diff
@@ -761,6 +783,15 @@ PointF NotationPaintView::toLogical(const QPoint& point) const
     return PointF::fromQPointF(m_matrix.inverted().map(scaledPoint.toQPoint()));
 }
 
+PointF NotationPaintView::fromLogical(const QPoint& point) const
+{
+    double scale = guiScaling();
```
Comment (njvdberg) [inline src/notation/view/notationpaintview.cpp:788]: This line might better placed a few lines lower, close the `return` statement where `scale` is actually used.

File: src/notation/view/notationpaintview.cpp [hunk 5]
```diff
@@ -761,6 +783,15 @@ PointF NotationPaintView::toLogical(const QPoint& point) const
...
+
+    PointF mappedPoint =  PointF::fromQPointF(m_matrix.map(point));
+
+    return mappedPoint / scale;
```
Comment (njvdberg) [inline src/notation/view/notationpaintview.cpp:792]: Maybe `return mappedPoint / guiScaling();` ?

File: src/notation/view/notationpaintview.cpp [hunk 6]
```diff
@@ -776,6 +807,16 @@ RectF NotationPaintView::toLogical(const QRect& rect) const
     return RectF::fromQRectF(m_matrix.inverted().mapRect(scaledRect));
 }
 
+RectF NotationPaintView::fromLogical(const QRect& rect) const
+{
+    double scale = guiScaling();
+
+    RectF mappedRect = RectF::fromQRectF(m_matrix.mapRect(rect));
+
+    mappedRect.setBottomRight(mappedRect.bottomRight() / scale);
```
Comment (njvdberg) [inline src/notation/view/notationpaintview.cpp:816]: Same comments as above, move `double scale = guiScaling()` a few lines down or remove the variable `scale` completely.

File: src/notation/qml/MuseScore/NotationScene/internal/ElementPopup.qml [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+import MuseScore.Inspector 1.0
+
+
+StyledPopup {
+    id: root
+
+    property var model;
+    property string type;
+
+    closePolicy: Popup.CloseOnPressOutside
```
Comment (njvdberg) [inline src/notation/qml/MuseScore/NotationScene/internal/ElementPopup.qml:18]: Is there any reason not to use `Popup.CloseOnEscape | Popup.CloseOnPressOutside` instead of `Popup.CloseOnPressOutside`?

File: src/notation/view/notationpaintview.cpp [hunk 8]
```diff
@@ -343,6 +343,28 @@ void NotationPaintView::showContextMenu(const ElementType& elementType, const QP
     emit openContextMenuRequested(menuItems, pos);
 }
 
+void NotationPaintView::showElementPopup(const Element* hitElement)
+{
+    if (!hitElement->needsElementPopup())
+        return;
+
+    const RectF& bbox = hitElement->canvasBoundingRect();
+    auto a = hitElement->pageBoundingRect();
```
Comment (njvdberg) [inline src/notation/view/notationpaintview.cpp:352]: Variable `a` is not used.

File: src/notation/view/notationpaintview.cpp [hunk 9]
```diff
@@ -343,6 +343,28 @@ void NotationPaintView::showContextMenu(const ElementType& elementType, const QP
...
+    auto b = hitElement->bbox();
```
Comment (njvdberg) [inline src/notation/view/notationpaintview.cpp:353]: Variable `b` is not used.

File: src/main/CMakeLists.txt (hunk 2 above)
Comment (cbjeukendrup) [reply src/main/CMakeLists.txt:520]: Sidharth explained to me earlier that Visual Studio adds the `--debug` flag when running the app. This prevented the app from running, because the flag was not recognized. Therefore you need to tell Visual Studio every time that it should not add that flag. Now, Sidharth apparently found the root cause of that flag. 

However, the app should now run properly with this flag since PR #8294. 
I don't know if this change should be there anyway (and if so, it should probably be at least in a different commit).

File: src/notation/qml/MuseScore/NotationScene/internal/ElementPopup.qml [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+import QtQuick.Layouts 1.15
+import QtQuick.Controls 2.15
+import QtQuick.Controls 1.3
+
+import MuseScore.NotationScene 1.0
+import MuseScore.UiComponents 1.0
+import MuseScore.Ui 1.0
+import MuseScore.Inspector 1.0
+
+
```
Comment (cbjeukendrup) [inline src/notation/qml/MuseScore/NotationScene/internal/ElementPopup.qml:11]: Please don't over-use newlines. Use at most one in a row, and don't put newlines before closing braces (see bottom of this file).

File: src/main/CMakeLists.txt (hunk 2 above)
Comment (sidharth-anand) [reply src/main/CMakeLists.txt]: Confirmed that this change is not required because PR #8294 adds the debug flag

Comment (njvdberg) [reply src/main/CMakeLists.txt]: Great! In that case I propose to remove this change and the rebase this PR.

**Review by njvdberg — APPROVED** [review]
Thanks!!

File: src/inspector/models/inspectorlistmodel.cpp [hunk 11]
```diff
@@ -63,7 +63,8 @@ void InspectorListModel::buildModelsForEmptySelection(const QSet<Ms::ElementType
 {
     static QList<AbstractInspectorModel::InspectorSectionType> persistentSectionList {
         AbstractInspectorModel::InspectorSectionType::SECTION_SCORE_DISPLAY,
-        AbstractInspectorModel::InspectorSectionType::SECTION_SCORE_APPEARANCE
+        AbstractInspectorModel::InspectorSectionType::SECTION_SCORE_APPEARANCE,
+        AbstractInspectorModel::InspectorSectionType::SECTION_NOTATION,
```
Comment (vpereverzev) [inline src/inspector/models/inspectorlistmodel.cpp:67]: I encourage you to exclude any changes for inspector from this PR

Comment (sidharth-anand) [reply src/inspector/models/inspectorlistmodel.cpp:67]: I've only made this change because ElementPopup (see NotationView.qml, line 121) needs the SECTION_NOTATION model to be available at startup

**Comment by Jojo-Schmitz** [comment]
Rebase needed

**Comment by RomanPudashkin** [comment]
Closing as the same feature has been implemented in [this PR](https://github.com/musescore/MuseScore/pull/12597) (which will be merged soon)
